Sources/Datadog/RUM/RUMTelemetry.swift

```diff
@@ -8,14 +8,27 @@ import Foundation
 
 /// Sends Telemetry events to RUM.
 ///
-/// `RUMTelemetry` complies to `Telemetry` protocol allowing
-/// sending telemetry events accross features.
+/// `RUMTelemetry` complies to `Telemetry` protocol allowing sending telemetry
+/// events accross features.
+///
+/// Events are reported up to 100 per sessions with a sampling mechanism that is
+/// configured at initialisation. Duplicates are discared.
 internal final class RUMTelemetry: Telemetry {
+    /// Maximium number of telemetry events allowed per user sessions.
+    static let MaxEventsPerSessions: Int = 100
+
     let sdkVersion: String
     let applicationID: String
     let source: String
     let dateProvider: DateProvider
     let dateCorrector: DateCorrectorType
+    let sampler: Sampler
+
+    /// Keeps track of current session
+    private var currentSessionID: RUMUUID = .nullUUID
+
+    /// Keeps track of event's ids recorded during a user session.
+    private var eventIDs: [String] = []
```

Comment:
`Set<String>` would be better - both in terms of performance and to express the purpose of `eventIDs`.